nncf/common/batchnorm_adaptation.py

```diff
@@ -0,0 +1,105 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Optional
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from nncf.api.compression import ModelType
+from nncf.common.initialization import NNCFDataLoader
+from nncf.common.utils.logger import logger as nncf_logger
+import nncf.common.factory as factory
+
+
+class BatchnormAdaptationAlgorithmImpl(ABC):
+    """
+    This is the class from which all framework-specific implementations of
+    the batch-norm adaptation algorithm inherit.
+    """
+
+    def __init__(self,
+                 data_loader: NNCFDataLoader,
+                 num_bn_adaptation_steps: int,
+                 num_bn_forget_steps: int,
+                 device: Optional[str] = None):
+        """
+        Initializes the batch-norm adaptation algorithm implementation.
```

Comment:
Please add docstring for the arguments because they have another meaning than arguments of the `BatchnormAdaptationAlgorithm`.